packages/shader-lab/src/lexer/Lexer.ts

```
@@ -432,6 +479,199 @@ export class Lexer extends BaseLexer {
+    // Peek the first word of the value. If it's a GLSL keyword that doesn't start
+    // expressions (type, qualifier, statement leader), the value isn't
+    // expression-shaped — use the legacy path. Non-keyword identifiers and the
+    // explicit expression-starter keywords (`true`/`false`) go through the AST path.
+    if (BaseLexer.isAlpha(src.charCodeAt(i))) {
+      const wordStart = i;
+      while (i < src.length && BaseLexer.isAlnum(src.charCodeAt(i))) i++;
+      const firstWord = src.substring(wordStart, i);
+      if (
+        Lexer._lexemeTable[firstWord] !== undefined &&
+        !Lexer._expressionValueLeaderKeywords.has(firstWord)
+      ) {
```

ESLint

[error] Replace `⏎········Lexer._lexemeTable[firstWord]·!==·undefined·&&⏎········!Lexer._expressionValueLeaderKeywords.has(firstWord)⏎······` with `Lexer._lexemeTable[firstWord]·!==·undefined·&&·!Lexer._expressionValueLeaderKeywords.has(firstWord)`

(prettier/prettier)